dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/preheat/mappers/OrganisationUnitMapperTest.java

```diff
@@ -58,4 +59,26 @@ void testIdSchemeRelatedFieldsAreMapped()
         assertEquals( "green", mapped.getCode() );
         assertContainsOnly( Set.of( attributeValue( "m0GpPuMUfFW", "purple" ) ), mapped.getAttributeValues() );
     }
+
+    @Test
+    void testParentFieldsAreMapped()
+    {
+        OrganisationUnit rootOrgUnit = new OrganisationUnit();
+        rootOrgUnit.setUid( "root" );
+        OrganisationUnit level1OrgUnit = new OrganisationUnit();
+        level1OrgUnit.setUid( "level1" );
+        OrganisationUnit level2OrgUnit = new OrganisationUnit();
+        level2OrgUnit.setUid( "level2" );
+
+        level2OrgUnit.setParent( level1OrgUnit );
+        level1OrgUnit.setParent( rootOrgUnit );
+        rootOrgUnit.setParent( null );
+
+        OrganisationUnit mapped = OrganisationUnitMapper.INSTANCE.map( level2OrgUnit );
+
+        assertEquals( "level2", mapped.getUid() );
+        assertEquals( "level1", mapped.getParent().getUid() );
```

Comment:
This can throw an NPE, so best to add a `assertNotNull(mapped.getParent());` before. Same below.